collect_app/src/androidTest/java/org/odk/collect/android/regression/FillBlankFormWithRepeatGroupTest.java

```diff
@@ -0,0 +1,197 @@
+package org.odk.collect.android.regression;
+
+import android.Manifest;
+
+import androidx.test.rule.GrantPermissionRule;
+import androidx.test.runner.AndroidJUnit4;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.espressoutils.pages.FormEntryPage;
+import org.odk.collect.android.espressoutils.pages.GeneralSettingsPage;
+import org.odk.collect.android.espressoutils.pages.MainMenuPage;
+import org.odk.collect.android.support.CopyFormRule;
+import org.odk.collect.android.support.ResetStateRule;
+
+//Issue NODK-247
+@RunWith(AndroidJUnit4.class)
+public class FillBlankFormWithRepeatGroupTest extends BaseRegressionTest {
+
+    @Rule
+    public RuleChain copyFormChain = RuleChain
+            .outerRule(GrantPermissionRule.grant(
+                    Manifest.permission.READ_EXTERNAL_STORAGE,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                    Manifest.permission.READ_PHONE_STATE)
+            )
+            .around(new ResetStateRule())
+            .around(new CopyFormRule("TestRepeat.xml"))
+            .around(new CopyFormRule("form1.xml"))
+            .around(new CopyFormRule("form2.xml"))
+            .around(new CopyFormRule("form3.xml"))
+            .around(new CopyFormRule("form4.xml"))
+            .around(new CopyFormRule("form5.xml"))
+            .around(new CopyFormRule("form6.xml"))
+            .around(new CopyFormRule("form7.xml"))
+            .around(new CopyFormRule("form8.xml"))
+            .around(new CopyFormRule("form9.xml"))
+            .around(new CopyFormRule("RepeatGroupAndGroup.xml"))
+            .around(new CopyFormRule("basic.xml"))
+            .around(new CopyFormRule("repeat_group_form.xml"));
+
+    @Test
+    public void repeatGroupsNotAdded_ShouldNotDoubleLastMessage() {
```

Comment:
Not sure I understand the name of this test. What's the "last message" here?

Reply:
I changed it to 'ShouldNotDoubleLastQuestion'. 
We had a problem where this last question was doubled when no group was added.

Reply:
Ah cool. Could it be `whenNoRepeatGroupAdded_doesNotDoubleLastQuestion`? I find adding `when` helps clarify it's the "conditional" in the test. Some people really like `should` but I find it is just noise - we're not hoping ("it should do this") we're stating ("it does do this").